spec/datadog/tracing/metadata/metastruct_tagging_spec.rb

```diff
@@ -0,0 +1,34 @@
+require 'spec_helper'
+
+require 'datadog/tracing/metadata/metastruct_tagging'
+
+RSpec.describe Datadog::Tracing::Metadata::MetastructTagging do
+  subject(:test_object) { test_class.new }
+
+  let(:test_class) do
+    Class.new { include Datadog::Tracing::Metadata::MetastructTagging }
+  end
+
+  describe '#set_metastruct_tag' do
+    it 'sets the metastruct to a hash with given key / value pair' do
+      test_object.set_metastruct_tag(:foo, [{ some: 'value' }])
+
+      expect(test_object.get_metastruct_tag(:foo)).to eq([{ some: 'value' }])
+    end
+
+    it 'does not lose previous entries' do
+      test_object.instance_variable_set(:@metastruct, { bar: [1] })
```

Comment:
I think it could be changed into a pair of `change` expectations instead

```ruby
expect { set_metastruct_tag(:bar, [1]) }.to change { get_metastruct_tag(:bar) }.from("...").to("...")

expect { set_metastruct_tag(:foo, [2]) }.not_to change { get_metastruct_tag(:bar) }.from([1]).and to change {}.from("...").to("...")

```